bftengine/src/bcstatetransfer/SourceSelector.cpp

```diff
@@ -175,7 +187,49 @@ void SourceSelector::selectSource(uint64_t currTimeMilli) {
   fetchingTimeStamp_ = 0;
   fetchRetransmissionCounter_ = 0;
   receivedValidBlockFromSrc_ = false;
+
+  LOG_INFO(logger_, "Selected new source replica " << currentReplica_);
+}
+
+void SourceSelector::updateCurrentPrimary(uint16_t newPrimary) {
+  if (currentPrimary_ == newPrimary) return;
+  ConcordAssert(isValidSourceId(newPrimary));
+  nominatedPrimaryCounter_++;
+
+  auto resetNominatedPrimary = [&]() {
+    nominatedPrimary_ = NO_REPLICA;
+    nominatedPrimaryCounter_ = 0;
+  };
+
+  if (nominatedPrimary_ == NO_REPLICA) {
+    nominatedPrimary_ = newPrimary;
+    return;
+  }
+
+  if ((newPrimary != nominatedPrimary_) and (nominatedPrimaryCounter_ < MAX_PREPREPARE_MSGS)) {
```

Comment:
I'm a bit confused here.
You still have MAX_PREPREPARE_MSGS but I saw you defined maxPrePrepareMessages as well.
Don't you think we should have it as a configuration parameter?
I lost my original comment so asking here again.